File: Resueltos_por_competidor/Miguel_Lopez/TwoGram.cpp
/*
 * Autor: Miguel Lopez
 * Problema: Two-Gram
 * Juez online: Codeforces
 * Veredicto: Accepted
 * Url: https://codeforces.com/contest/977/problem/B
 */

#include <bits/stdc++.h>
#include <cctype>
#include <cstdio>
#include <ios>
#include <ostream>
typedef long long int ll;
using namespace std;
void init_code() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    #ifndef ONLINE_JUDGE
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    #endif 
}
int main() {
    init_code();
    int t; cin >> t;
    string s; cin >> s;
    map<string, int> freq;
    for(int i = 0; i+1 < s.length(); i++) {
        string str = s.substr(i,2);
        freq[str]++;
    }
    string str = "";
    int cont = 0;
    for(auto i: freq) {
        if(i.second > cont) {
            cont = i.second;
            str = i.first;
        }
    }
    cout << str << endl;
}

File: Resueltos_por_competidor/Martin_Peña/TwoGram.java
/*
* Autor: pgms1103-afk
* Problema: B. Two-gram
* Juez Online: Codeforces
* Veredicto: Accepted
* URL: https://codeforces.com/contest/977/problem/B
*/
import java.util.HashMap;
import java.util.Scanner;

public class TwoGram {
	static Scanner sc = new Scanner (System.in);
	public static void main(String[] args) {
		
		sc.nextInt();
		String s = sc.next();
		
		HashMap <String, Integer> subStrings = new HashMap<>();
		int mayor = 0;
		for (int i = 0; i < s.length()-1; i++) {
			subStrings.merge(""+s.charAt(i) + s.charAt(i+1), 1, Integer::sum);
		}
		
		Integer[] valores = new Integer[subStrings.size()];
		subStrings.values().toArray(valores);
	
		for (int i = 0; i < subStrings.size(); i++) {
			if(valores[i] > mayor) {
				mayor = valores[i];
			}
		}
		
		for(String key: subStrings.keySet()) {
			if(subStrings.get(key) == mayor) {
				System.out.println(key);
				break;
			}
		}
		
		
		
		
		
		
	}

}
